packages/core/src/sync-realtime/index.ts

```diff
@@ -453,34 +453,71 @@ export const createRealtimeSync = (
       chainId: args.network.chainId,
     });
 
-    // Remove logs that don't match a filter
-    logs = logs.filter((log) =>
-      logFilters.some((filter) => isLogFilterMatched({ filter, block, log })),
-    );
+    /**
+     * key: matched transaction hashes
+     * value: `true` if the corresponding receipt should be fetched
+     */
+    const transactionHashes = new Map<Hash, boolean>();
```

Comment:
I think this would be more readable as two `Set<Hash>` named `requiredTransactionInputs` and `requiredTransactionReceipts`

Reply:
Agree, was wondering whether duplicating data in two sets would make sense for this purpose.